lib/system/channels.nim

```diff
@@ -15,6 +15,98 @@
 ## not work with cyclic data structures.
 ## **Note:** Channels cannot be passed between threads. Use globals or pass
 ## them by `ptr`.
+##
+## Example
+## =======
+## In this example, a program is created that uses the ``threadpool`` module
+## to launch two separate procedures. These procedures send strings through a
+## channel to the main thread, which calls ``echo`` on the strings.
+## Both blocking and non-blocking receives are demonstrated.
+## The channel is declared as a global to avoid the unsafety of
+## passing it by raw ``ptr``.
+##
+## .. code-block :: Nim
+##   # For simple and efficient multithreading.
```

Comment:
Any particular reason there send to be a self-imposed 50-character line length here? Usually 80 is used.

Reply:
Probably VSCode :)

I'll need to adjust the over-length guard line.